node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -335,23 +329,53 @@ object WebApi {
   private def exprFromBundleProto(b: Bundle): Option[RhoExpr] = exprFromParProto(b.body)
 
   private def mkUnforgExpr(f: String => RhoUnforg, bs: ByteString): ExprUnforg =
-    ExprUnforg(f(toHex(bs)))
+    ExprUnforg(f(bs.toHexString))
 
   // RhoExpr to protobuf
 
+  private def rhoExprToParProto(exp: RhoExpr): Par = {
+    def exprProtoToParProto(exp: Expr): Par = Par(exprs = Seq(exp))
+
+    exp match {
+      // Nested expressions (Par, Tuple, List and Set are converted to JSON list)
+      case ExprPar(data) =>
+        exprProtoToParProto(Expr().withEListBody(EList(data.map(rhoExprToParProto))))
```

Comment:
This is not correct, ExprPar represents Par in proto, it's not the same as Expr List.

Reply:
@tgrospic, I'm not sure how to correctly fix it. As far as I remember `Par` can contain nested `Par`s in `Bundle` field. So I'm trying to reqursively build `Par` from nested `Par`s as sequence of Bundles.
```scala
case ExprPar(data) => Par(bundles = data.map(expr => Bundle(rhoExprToParProto(expr))))
```

Reply:
Please take a look [8f55e85](https://github.com/rchain/rchain/pull/3718/commits/8f55e8563a403c128564d58d788c8aaba630ccd7).